tools/run_parallel.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Uncommenting the next line may be useful when debugging this script.
+#set -x
+
+usage () {
+    cat << EOF
+Usage: run_parallel.sh {command} {num_iter} {num_parallel}
+Where:
+  {command} is a string containing the command, including parameters, to run
+  {num_iter} is an positive integer indicating how many iterations to execute
+  {num_parallel} is an (optional) positive integer indicating how many parallel commands should be executed
+     in each iteration. If not provided, the default is half the number of available CPU cores.
+EOF
+}
+
+if [ "$#" -lt  2 ]; then
+    echo "Illegal number of parameters."
+    usage
+    exit 1
+fi
+
+# Calculate the number of CPU cores. This code is Linux specific at the moment.
+NCORES=$(echo "`grep -c ^processor /proc/cpuinfo` / 2" | bc)
+
+command=$1
+num_iter=$2
+
+if [ "$#" -eq  3 ]; then
+  num_parallel=$3
+else
+  num_parallel=$NCORES
+fi
+
+echo "run_parallel:"
+echo "  number of cores: $NCORES"
+echo "  command:         $command"
+echo "  num_parallel:    $num_parallel"
+echo "  num_iter:        $num_iter"
+
+outf=./outfile.txt
+
+for i in $(seq 1 $num_iter); do
```

Comment:
Is `1` appending `seq` necessary?

Reply:
hi @lukech , the reason I started `i` at 1 (rather than 0) because it seemed more natural for the line `echo "Starting iteration $i"` to count from 1 to n, rather than from 0 to (n-1).

Reply:
Not sure if it's about the shell version. My bash (version 3.2.57) gave me 1-5 (instead of 0-4) if omitting the `1` right after `seq`.

Reply:
Hi @lukech , I understand now, and you're right. Using `seq` with two parameters `n` and `m` gives you the sequence from n to m inclusive. If you only provide one parameter `x`, then you get the sequence from 1 to x inclusive. So, in this case, the behaviour of the current code is correct, but the `1` is not required.

Reply:
Hi @lukech , I've removed the redundant `1`.